doc/cffi-manual.texinfo

```diff
@@ -6983,13 +6983,22 @@ invocation.
 @node Static Linking, Limitations, The Groveller, Top
 @chapter Static Linking
 
-On recent enough versions of supported implementations (currently,
-GNU CLISP 2.49, CMUCL 2015-11, and SBCL 1.2.17), and with a recent
-enough ASDF (3.1.2 or later), you can create a statically linked
-Lisp executable image that includes all the C extensions
-(wrappers and any other objects output by @code{compile-op})
-as well as your Lisp code --- or a standalone application executable.
-This makes it easier to deliver your code as a single file.
+@impnote{``Static Linking'' here does not mean the same as in the C
+world. It does not build an ELF static binary. It builds an image or
+program where all the libraries built by CFFI (grovel or wrapper
+files) are statically embedded in the image or program. The resulting
```

Comment:
the groveller builds and runs a C program, but it's only used at compile time. It's not bundled by static-image-op (is it?).

Reply:
As far as I understand, it is. See e.g. https://github.com/cffi/cffi/blob/master/toolchain/c-toolchain.lisp#L275

Reply:
I don't understand what that line does or in what context it's executed. :-( IIUC, static-foo-op will display the ld commands that perform static linking. Did you see it statically linking a grovel file? They have a main() function, doesn't that cause problems?

Reply:
Here is what appears for e.g. osicat (with some newlines for clarity) (where my system is named "ballish-daemon":

```
# compiling an object file
; cc \
    -o /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/dists/quicklisp/software/osicat-20200427-git/posix/libosicat.so \
    -shared -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer \
    /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/dists/quicklisp/software/osicat-20200427-git/posix/wrappers__wrapper.o

# building a static library using the previously compiled object file
; ar rcsDT \
    /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/builds/ralt/ballish/ballish-daemon-tmpP07HWIXP.a \
    /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/dists/quicklisp/software/osicat-20200427-git/posix/wrappers__wrapper.o

# compiling a runtime image with the static library
; cc \
    -o /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/builds/ralt/ballish/ballish-runtime-tmpURW0SA3C \
    -g -Wl,--export-dynamic /usr/local/lib/sbcl/sbcl.o \
    -Wl,--whole-archive /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/builds/ralt/ballish/ballish-daemon.a \
    -Wl,--no-whole-archive -ldl -lpthread -lz -lm -lm
```

And the next line is running a dumping command using the new runtime binary.


Another dependency in that project is cl-inotify, which is using the groveller:

```
; cc -o /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel-tmpQ371UGST.o -c -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fPIC -I/home/lisp/quicklisp/local-projects/cffi/ /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel.c

; cc -o /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel-tmp2OWI3Q7U -g -Wl,--export-dynamic /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel.o

; /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel /home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel__grovel.grovel-tmp.lisp

; compiling file "/home/lisp/.cache/common-lisp/sbcl-2.0.3-linux-x64/home/lisp/quicklisp/local-projects/cl-inotify/src/grovel.processed-grovel-file" (written 20 MAY 2020 12:36:42 PM):
```

Which seems to compile a binary (grovel__grovel), then running that binary to generate this grovel.processed-grovel-file, which is a Lisp file (full of `cl:defconstant` in this example).

Reply:
Right, so your note shouldn't imply that grovel files are statically linked into the lisp image, correct?

Reply:
So I guess I should change the doc to say "wrapper files" only, not grovel.

Reply:
... your comment hadn't appeared when I posted mine, but yes, we agree :) will do so and ping you again.

Reply:
@luismbo updated. Looks better?